v24.1.0-rc.2 release notes (#18547)
cockroachdb/docs: src/current/_includes/releases/v24.1/v24.1.0-rc.2.md

## v24.1.0-rc.2

Release Date: May 16, 2024

{% include releases/new-release-downloads-docker-image.md release=include.release %}

<h3 id="v24-1-0-rc-2-bug-fixes">Bug fixes</h3>

> - Fixed a bug that was present since v22.2 where [changefeeds]({% link v24.1/change-data-capture-overview.md %}) with long-running [initial scans]({% link v24.1/create-changefeed.md %}#initial-scan) may incorrectly restore checkpoint job process and drop events during [changefeed restarts]({% link v24.1/changefeed-messages.md %}#duplicate-messages) due to transient errors or node restarts. The bug was most likely to occur in clusters with the following contributing factors:

Reviewer: ```suggestion
- Fixed a bug that was present since v22.2 where [changefeeds]({% link v24.1/change-data-capture-overview.md %}) with long-running [initial scans]({% link v24.1/create-changefeed.md %}#initial-scan) might incorrectly restore checkpoint job progress and drop events during [changefeed restarts]({% link v24.1/changefeed-messages.md %}#duplicate-messages) due to transient errors or node restarts. The bug was most likely to occur in clusters with the following contributing factors:
```

I prefer "might" because "may" is more definitively future tense despite this state (former bug) existing in the past.

I am suggesting process -> progress because this matches the original release note and works better gramatically but if it was changed intentionally and it's more accurate, feel free to ignore this.

Author: Cool, oh yeah, thanks for catching "process" is definitely a typo.

    - The `changefeed.shutdown_checkpoint.enabled` [cluster setting]({% link v24.1/cluster-settings.md %}) was enabled (in clusters running v23.2 and later).

>     - The cluster settings `changefeed.frontier_checkpoint_frequency` and `low changefeed.frontier_highwater_lag_checkpoint_threshold` were set low, which result in the initial scan taking many multiples of the configured frequency to complete.

Reviewer: ```suggestion
    - The cluster settings `changefeed.frontier_checkpoint_frequency` and `low changefeed.frontier_highwater_lag_checkpoint_threshold` were set low, which resulted in the initial scan taking many multiples of the configured frequency to complete.
```

    - There were multiple target tables with significant differences in row counts in one changefeed.
    - The changefeed target tables were large with many ranges.
    - The initial scan took a long time to complete (an hour or longer). [#123968][#123968]

> - Fixed a crash introduced in v24.1.0-beta.2 that could occur when planning [statistics collection]({% link v24.1/cost-based-optimizer.md %}#table-statistics) on a table with a [virtual computed column]({% link v24.1/computed-columns.md %}) using a user-defined type and the `sql.stats.virtual_computed_columns.enabled` [cluster setting]({% link v24.1/cluster-settings.md %}) is set to `true`. `sql.stats.virtual_computed_columns.enabled` was introduced in v24.1.0-alpha.1 by default enabled. [#124064][#124064]

Reviewer: ```suggestion
- Fixed a crash introduced in v24.1.0-beta.2 that could occur when planning [statistics collection]({% link v24.1/cost-based-optimizer.md %}#table-statistics) on a table with a [virtual computed column]({% link v24.1/computed-columns.md %}) using a user-defined type and the `sql.stats.virtual_computed_columns.enabled` [cluster setting]({% link v24.1/cluster-settings.md %}) is set to `true`. `sql.stats.virtual_computed_columns.enabled` was introduced in v24.1.0-alpha.1 as `true` by default and introduced in v23.2.5 as `false` by default. [#124064][#124064]
```

I'd want to improve the wording of the original, regardless, but the additional 23.2 bit might also prevent confusion for someone who had the setting in 23.2 and doesn't understand why we seem to claim to have 'introduced' it in 24.1

<div class="release-note-contributors" markdown="1">

<h3 id="v24-1-0-rc-2-contributors">Contributors</h3>

This release includes 3 merged PRs by 3 authors.

> </div>

Reviewer: I believe I've removed this in the past when there are no community contributions. May be too awkward/confusing to leave it in when there are only 2 linked PRs here? I could also understand leaving it, if readers have grown accustomed to having a count.
